surveyor.py

```diff
@@ -325,11 +325,11 @@ def survey(ctx, product: str = 'cbr'):
         if opt.ioc_file:
             with open(opt.ioc_file) as ioc_file:
                 data = ioc_file.readlines()
-                log_echo(f"Processing IOC file: {ioc_file}", log)
+                log_echo(f"Processing IOC file: {opt.ioc_file}", log)
 
                 for ioc in data:
                     ioc = ioc.strip()
-                    product.nested_process_search(Tag(ioc), {opt.ioc_type: [ioc]}, base_query)
+                    product.nested_process_search(Tag(ioc, opt.ioc_file), {opt.ioc_type: [ioc]}, base_query)
 
                 for tag, results in product.get_results().items():
                     _write_results(writer, results, ioc, 'ioc', tag, log)
```

Comment:
I suggest we model this like how the def_file is processed. I was hoping to find a way to filter in the product file by the tag but was struggling to find a solution. The hash ioctype currently gets stuck in the regex check and formatted incorrectly. Some changes for hashes will need to be pushed to product files along with proper parameter mappings otherwise this code works great!

```
            with open(opt.ioc_file) as ioc_file:
                basename = os.path.basename(opt.ioc_file)
                data = ioc_file.readlines()
                log_echo(f"Processing IOC file: {opt.ioc_file}", log)

                for ioc in data:
                    ioc = ioc.strip()
                    product.nested_process_search(Tag('ioc', data=basename), {opt.ioc_type: [ioc]}, base_query)
```

Reply:
One idea to avoid the regex confusion - we could adjust the `survyeor.py` code to be:
```
            with open(opt.ioc_file) as ioc_file:
                basename = os.path.basename(opt.ioc_file)
                data = ioc_file.readlines()
                log_echo(f"Processing IOC file: {opt.ioc_file}", log)

                for ioc in data:
                    ioc = ioc.strip()
                    product.nested_process_search(Tag(f"IOC - {ioc}", data=basename), {opt.ioc_type: [ioc]}, base_query)
```

Then, in `sentinel_one.py`, we can key off the tag name so that the logic can look something like:

```
                if parameter == 'query':
                    # Formats queries as (a) OR (b) OR (c) OR (d)
                    if len(terms) > 1:
                        search_value = ') OR ('.join(terms)
                    else:
                        search_value = terms[0]
                    operator = 'raw'
                elif len(terms) > 1:
                    search_value = f'({all_terms})'
                    operator = 'in contains anycase'
                elif not re.findall(r'\w+\.\w+', search_value) and tag.tag != f"IOC - {search_value[0]}":
                    operator = 'regexp'
                else:
                    operator = 'containscis'
```

We can also include the parameter mapping entry for SentinelOne as an intermediate solution until there's a PR for #65 
```
{
  "md5": "Md5"
}
```